src/state/slices/opportunitiesSlice/utils.ts

```diff
@@ -74,21 +75,23 @@ export const getUnderlyingAssetIdsBalances: GetUnderlyingAssetIdsBalances = ({
   underlyingAssetRatiosBaseUnit,
   cryptoAmountBaseUnit,
   assets,
+  assetId,
   marketData,
 }) => {
   return Object.values(underlyingAssetIds).reduce<GetUnderlyingAssetIdsBalancesReturn>(
     (acc, underlyingAssetId, index) => {
       const underlyingAsset = assets[underlyingAssetId]
+      const asset = assets[assetId ?? '']
       const marketDataPrice = marketData[underlyingAssetId]?.price
       if (!underlyingAsset) return acc
       const fiatAmount = bnOrZero(cryptoAmountBaseUnit)
         .times(fromBaseUnit(underlyingAssetRatiosBaseUnit[index], underlyingAsset.precision))
-        .div(bnOrZero(10).pow(underlyingAsset?.precision))
+        .div(bnOrZero(10).pow(asset?.precision ?? underlyingAsset.precision))
         .times(marketDataPrice ?? 0)
         .toString()
       const cryptoBalancePrecision = bnOrZero(cryptoAmountBaseUnit)
         .times(fromBaseUnit(underlyingAssetRatiosBaseUnit[index], underlyingAsset.precision))
-        .div(bnOrZero(10).pow(underlyingAsset?.precision))
+        .div(bnOrZero(10).pow(asset?.precision ?? underlyingAsset?.precision))
         .toString()
```

Comment:
switch these around and only do calculation once?

Reply:
d22032aaaa9419b3650f2e9563723a7025fc6702